pwiz_tools/Skyline/Model/Proteome/ProteinGroupMetadata.cs

```diff
@@ -0,0 +1,216 @@
+ *
+ * Copyright 2022
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using pwiz.Common.Collections;
+using pwiz.Common.SystemUtil;
+using pwiz.ProteomeDatabase.API;
+using pwiz.Skyline.Util;
+
+namespace pwiz.Skyline.Model.Proteome
+{
+    public class ProteinGroupMetadata : ProteinMetadata
+    {
+        const string GROUP_SEPARATOR = @"/";
+        public new static readonly ProteinGroupMetadata EMPTY = new ProteinGroupMetadata();
+
+        public ImmutableList<ProteinMetadata> ProteinMetadata { get; }
+
+        [Track]
+        public override string Name => ProteinMetadata.All(p => p.Name == null)
+            ? null
+            : string.Join(GROUP_SEPARATOR, ProteinMetadata.Select(p => p.Name));
+
+        [Track]
+        public override string Description => ProteinMetadata.All(p => p.Description == null)
+            ? null
+            : string.Join(GROUP_SEPARATOR,
+                ProteinMetadata.Where(p => !string.IsNullOrWhiteSpace(p.Description)).Select(p => p.Description));
+
+        [Track]
+        public override string PreferredName => ProteinMetadata.All(p => p.PreferredName == null)
+            ? null
+            : string.Join(GROUP_SEPARATOR,
+                ProteinMetadata.Where(p => !string.IsNullOrWhiteSpace(p.PreferredName)).Select(p => p.PreferredName));
+
+        [Track]
+        public override string Accession => ProteinMetadata.All(p => p.Accession == null)
+            ? null
+            : string.Join(GROUP_SEPARATOR,
+                ProteinMetadata.Where(p => !string.IsNullOrWhiteSpace(p.Accession)).Select(p => p.Accession));
+
+        [Track]
+        public override string Gene => ProteinMetadata.All(p => p.Gene == null)
+            ? null
+            : string.Join(GROUP_SEPARATOR,
+                ProteinMetadata.Where(p => !string.IsNullOrWhiteSpace(p.Gene)).Select(p => p.Gene));
+
+        [Track]
+        public override string Species => ProteinMetadata.All(p => p.Species == null)
+            ? null
+            : string.Join(GROUP_SEPARATOR,
+                ProteinMetadata.Where(p => !string.IsNullOrWhiteSpace(p.Species)).Select(p => p.Species));
+        [Track]
+        public override WebSearchInfo WebSearchInfo => ProteinMetadata.Select(p => p.WebSearchInfo).First();
+
+        public override ProteinMetadata ChangeName(string name)
+        {
+            return new ProteinGroupMetadata(this);
```

Comment:
I think I put these in as placeholders planning to come back to it. Should non-protein-group-aware Skyline code be able to change protein groups' metadata properties? Or should they always be calculated from the constituent protein metadata (e.g. Protein1/Protein2/Protein3)?

Reply:
I think the latter. Certainly would be strange for a user to change WebSearch on ProteinMetadata of any kind, since that is a hidden field. This maybe explains why the code fails TestAuditLogLocalization complaining about WebSearch not being in PropertyNames.resx